### PR TITLE
Nez-100 Allow caching of Parser (btwn Threads) and optimize populating materialized views

### PR DESCRIPTION
**Pull request by amoghmargoor**

### CODE REVIEW

File: optimizer/src/main/java/org/apache/calcite/plan/QuarkMaterializeCluster.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   */
+  QuarkMaterializeCluster(RelOptPlannerHolder plannerHolder,
+                          RelDataTypeFactory typeFactory,
+                          RexBuilder rexBuilder, AtomicInteger nextCorrel,
+                          Map<String, RelNode> mapCorrelToRel) {
+    super(plannerHolder.getPlanner(), typeFactory, rexBuilder, nextCorrel, mapCorrelToRel);
+    this.plannerHolder = plannerHolder;
+  }
+  /** Creates a cluster. */
+  public static QuarkMaterializeCluster create(RelOptPlannerHolder plannerHolder,
```
Comment (vrajat) [inline optimizer/src/main/java/org/apache/calcite/plan/QuarkMaterializeCluster.java]: Why is this a static and not another constructor that calls the constructor in line 48 ?

Comment (amoghmargoor) [inline optimizer/src/main/java/org/apache/calcite/plan/QuarkMaterializeCluster.java]: Just followed RelOptCluster as they do the same thing to hide above constructor. Yeah they could have very well replaced it by another constructor too.

**Comment by vrajat** [comment]
Is this PR still relevant ? Can it be closed ?

**Comment by amoghmargoor** [comment]
yeah closing it.

**Pull request closed without merging**